napps/kytos/of_core/main.py

```diff
@@ -145,53 +147,101 @@ def handle_echo_request(self, event):
         """
         log.debug("Echo Request message read")
 
-        echo_request = event.content['message']
+        echo_request = event.message
         echo_reply = EchoReply(xid=echo_request.header.xid)
         event_out = KytosEvent(name=('kytos/of_core.messages.out.'
                                      'ofpt_echo_reply'),
                                content={'message': echo_reply,
                                         'destination': event.source})
         self.controller.buffers.msg_out.put(event_out)
 
+    @listen_to('kytos/core.connection.new')
+    def handle_core_new_connection(self, event):
+        """Method called when a new switch is connected.
+
+        This method will send a hello world message when a new switch is
+        connected.
+        """
+        self.say_hello(event.source)
```

Comment:
@macartur the Hello "response" should have the same `xid` from the Hello "request", doesn't it?